prow/cmd/phaino/local.go

```diff
@@ -246,23 +248,108 @@ func convertToLocal(ctx context.Context, log *logrus.Entry, pj prowapi.ProwJob,
 		log.Warn("Ignoring resource requirements")
 	}
 
+	// A map of volume mounts for the run.
+	// Key is the mount path and value is the local path.
+	volumeMounts := make(map[string]string)
 	for _, mount := range container.VolumeMounts {
+		if strExists(opt.skippedVolumesMounts, mount.Name) {
+			logrus.Infof("Volume mount %q skipped", mount.Name)
+			continue
+		}
 		vol := volume(*pj.Spec.PodSpec, mount.Name)
 		if vol == nil {
 			return nil, fmt.Errorf("mount %q missing associated volume", mount.Name)
 		}
 		if vol.EmptyDir != nil {
-			localArgs = append(localArgs, "-v", mount.MountPath)
+			volumeMounts[mount.MountPath] = ""
 		} else {
 			local, err := readMount(ctx, mount)
 			if err != nil {
 				return nil, fmt.Errorf("bad mount %q: %v", mount.Name, err)
 			}
-			arg := local + ":" + mount.MountPath
+			mountPath := mount.MountPath
 			if mount.ReadOnly {
-				arg += ":ro"
+				mountPath += ":ro"
+			}
+			volumeMounts[mountPath] = local
+		}
+	}
+	for pathInContainer, localPath := range opt.extraVolumesMounts {
+		volumeMounts[pathInContainer] = localPath
+	}
+
+	// A map of env vars for the run.
+	// Key is the env name and value is the env value.
+	envs := make(map[string]string)
+	for _, env := range container.Env {
+		if strExists(opt.skippedEnvVars, env.Name) {
+			continue
+		}
+		envs[env.Name] = env.Value
+	}
+	for name, value := range opt.extraEnvVars {
+		envs[name] = value
+	}
+	// Overwrite GOPATH in the container.
+	envs["GOPATH"] = opt.gopath
```

Comment:
Shouldn't we only do this when it `!= ""` 

I'm also unhappy that we're baking GOPATH so deeply into everything when it is an archaic golang feature.

Reply:
I reconsidered about this, it looks we don't actually need to overwrite the GOPATH, so I removed this line.